src/components/WebCLI/WebCLI.tsx

```diff
@@ -132,16 +135,38 @@ const WebCLI = ({
       onclose: () => {
         // Unused handler.
       },
+      onerror: (error) => {
+        setInlineError(
+          InlineErrors.CONNECTION,
+          typeof error === "string" ? error : "Unknown error.",
+        );
+      },
       messageCallback: (message: string) => {
         wsMessageStore.current = wsMessageStore.current + message;
         setOutput(wsMessageStore.current);
       },
     }).connect();
-    setConnection(conn);
+    setConnection((prevConnection) => {
+      if (!prevConnection || prevConnection.isOpen()) {
+        prevConnection?.disconnect();
+      } else {
+        setPendingConnections((prevPendingConnections) => [
+          ...prevPendingConnections,
+          prevConnection,
+        ]);
+      }
```

Comment:
Another option would be to just close the previous connection without checking if it is opened. This way, we won't need the extra `pendingConnections` state and all the extra logic surrounding it. The downside is that sometimes we might close the connection before it is opened and we'll receive a warning regarding this issue in the console.

Reply:
Removed `pendingConnections` state.